modules/common/condition/types.go

```diff
@@ -19,111 +19,65 @@ limitations under the License.
-		existingCondition := (*conditions)[i]
-		if existingCondition.Type == c.Type {
-			exists = true
-			if !hasSameState(&existingCondition, c) {
-				(*conditions)[i] = *c
-				break
-			}
-			c.LastTransitionTime = existingCondition.LastTransitionTime
-			break
-		}
-	}
-
-	// If the condition does not exist, add it, setting the transition time only if not already set
-	if !exists {
-		*conditions = append(*conditions, *c)
-	}
-}
-
-// hasSameState returns true if a condition has the same state of another
-func hasSameState(i, j *Condition) bool {
-	return i.Type == j.Type &&
-		i.Status == j.Status &&
-		i.Reason == j.Reason &&
-		i.Message == j.Message
+// Condition defines an observation of a API resource operational state.
+type Condition struct {
+	// Type of condition in CamelCase.
+	Type Type `json:"type"`
+
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+
+	// Severity provides a classification of Reason code, so the current situation is immediately
+	// understandable and could act accordingly.
+	// It is meant for situations where Status=False and it should be indicated if it is just
+	// informational, warning (next reconciliation might fix it) or an error (e.g. DB create issue
+	// and no actions to automatically resolve the issue can/should be done).
+	// +optional
+	Severity Severity `json:"severity,omitempty"`
+
+	// Last time the condition transitioned from one status to another.
+	// This should be when the underlying condition changed. If that is not known, then using the time when
+	// the API field changed is acceptable.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// The reason for the condition's last transition in CamelCase.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// A human readable message indicating details about the transition.
+	// +optional
+	Message string `json:"message,omitempty"`
 }
 
-// GetCurrentCondition - Get current condition with status == corev1.ConditionTrue
-func (conditions List) getCurrentCondition() *Condition {
-	for i, cond := range conditions {
-		if cond.Status == corev1.ConditionTrue {
-			return &conditions[i]
-		}
-	}
+// Conditions provide observations of the operational state of a API resource.
+type Conditions []Condition
 
-	return nil
-}
-
-// UpdateCurrentCondition - sets new condition, and sets previous condition to corev1.ConditionFalse
-func (conditions *List) UpdateCurrentCondition(c Condition) {
-	// if it is an empty condition, just return and don't set it
-	if c == (Condition{}) {
-		return
-	}
-
-	//
-	// get current condition and update to corev1.ConditionFalse
-	//
-	currentCondition := conditions.getCurrentCondition()
-	if currentCondition != nil {
-		currentCondition.Status = corev1.ConditionFalse
-		conditions.set(currentCondition)
-	}
-
-	//
-	// set new condition
-	//
-	conditions.set(&c)
-}
+// List - A list of conditions
+type List []Condition
```

Comment:
do we need this? I see type Conditions used in the new code.

Reply:
no, this is a left over

Reply:
done